src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/entity/Licence.kt

```diff
@@ -148,46 +156,50 @@ abstract class Licence(
     prisonCode: String,
     prisonDescription: String,
     prisonTelephone: String?,
-    updatedByUsername: String?,
+    staffMember: Staff?,
   ): Licence
 
-  fun updateAppointmentAddress(appointmentAddress: String?, updatedByUsername: String?) {
+  fun updateAppointmentAddress(appointmentAddress: String?, staffMember: Staff?) {
     this.appointmentAddress = appointmentAddress
     this.dateLastUpdated = LocalDateTime.now()
-    this.updatedByUsername = updatedByUsername
+    this.updatedByUsername = staffMember?.username ?: SYSTEM_USER
```

Comment:
I'm not sure if we need to default to `SYSTEM_USER` here, as I don't think there's a scenario where the system can update this info.
The only time I can see it being set to `SYSTEM_USER` would be if for some reason the `staffMember` doesn't exist, in which case we should probably know

Reply:
Had a chat with Andy about this and will seek clarification, we are working on these rules around where username is used and the `updated_by_id` column needs to be populated:
```
If you can't find a user, set the username to SYSTEM_USER
If you have a user, set the username to their username

If you can't find a user, leave the updated_by_id as it was and do not update
If you have a user, set the updated_by_id to the id of the user
```

Happy to jump on a call about this.

Reply:
Confirmed, logic should apply in all places we are setting username

Reply:
I think if we generally support those rules that should be fine - allow providing an optional updater rather than mandating that there is one.

Probably simpler to be consistent - at the small risk of the possibility of bugs if the username is not passed through.